### PR TITLE
Remove all supporting Web Portal code and nomenclature

### PR DESCRIPTION
**Pull request by jakeisonline**
This PR removes all code related to the support of the now sunset "Web Portal". It also removes references to previous nomenclature, namely "Entry" becoming "Article".

Note: The setting `nb_entries` remains as is, the name was not changed.

#### Risks

* Medium: This reaches deep into the core app code to rename key functions
* Support notes: Whilst the functionality does not change, the underlying behavior does. Even if an account has switched to Help Center, this app would have still be looking at Web Portal data. This app did not look at the account setting, relying instead on a manual toggle within the app's settings.
* QA notes: Manual testing performed, functionality seems to be normal after changes.

### CODE REVIEW

**Comment by pmgrove** [comment]
cc: @zendesk/vegemite

File: translations/ar.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
...
-  "no_entries": "لا توجد مقترحات لهذه التذكرة.",
+  "no_articles": "لا توجد مقترحات لهذه التذكرة.",
```
Comment (sandlerr) [inline translations/ar.json]: cc @zendesk/i18n this is a renamed string

File: app.js [hunk 2]
```diff
@@ -234,75 +215,40 @@
...
-      var topics = data.results,
-          topicIds = _.map(topics, function(topic) { return topic.id; });
-
-      this.ajax('fetchTopicsWithForums', topicIds)
-        .done(function(data){
-          var entries = this.formatEntries(topics, data);
-          this.store('entries', entries);
-          this.renderList(entries);
-        });
+      console.log("data:", data);
```
Comment (sandlerr) [inline app.js]: this needs to go, otherwise validation will fail

**Comment by sandlerr** [comment]
looks good! Since its a fairly big change let's deploy to staging and get @alieu 's review?

File: translations/ar.json (hunk 1 above)
Comment (jakeisonline) [inline translations/ar.json]: None of this goes into Rosetta, so not sure it really matters.

File: app.js [hunk 3]
```diff
@@ -383,16 +317,16 @@
...
+    numberOfDisplayableArticles: function(){
       return this.setting('nb_entries') || this.defaultNumberOfEntriesToDisplay;
     },
 
     queryLimit: function(){
       // ugly hack to return more results than needed because we filter out agent only content
-      if (this.setting('exclude_agent_only') && !this.setting('search_hc')) {
-        return this.numberOfDisplayableEntries() * 2;
+      if (this.setting('exclude_agent_only')) {
+        return this.numberOfDisplayableArticles() * 2;
```
Comment (adammw) [inline app.js]: Do we still need this 'ugly Jack's h seems like it wasn't being used for hc

File: app.js (hunk 2 above)
Comment (jakeisonline) [inline app.js]: Woops! Nice catch, `zat validate` didn't pick that up fwiw.

File: translations/ar.json (hunk 1 above)
Comment (ocke) [inline translations/ar.json]: I think these files are generated from rosetta. If you change them, zat translate will pull from rosette, and overwrite this back again.

Comment (jakeisonline) [inline translations/ar.json]: @ocke woah, I had no idea there is now a whole process involved in strings for apps. That's cool. So, rename string back to what it was, or have Rosetta/ZAT translate adjust?

File: app.js (hunk 3 above)
Comment (jakeisonline) [inline app.js]: What's an `ugly jack h`? It looks similar to things I've been called :)

Comment (adammw) [inline app.js]: *"ugly hack"

the perils of reviewing a PR on mobile without a bluetooth keyboard

**Comment by ydrain** [comment]
@zendesk/i18n-dev

**Comment by jakeisonline** [comment]
@zendesk/i18n-dev I reverted changes to any string keys. You're welcome to review this PR, but don't think its contents are applicable to i18n anymore.

**Comment by pdeuter** [comment]
+1 all good from i18n perspective

**Comment by sandlerr** [comment]
don't some of these strings need to be obsoleted? The parameters related to web portal are going away, they would have been translated.

**Comment by jakeisonline** [comment]
@sandlerr Sure, you can clean that up if you'd like! Along with all the logic for agent restricted content. Scope is just to remove the Web Portal lookup ultimately.

**Comment by sandlerr** [comment]
alright, I'll take over merging and shipping this then? Any urgency?

**Comment by jakeisonline** [comment]
@sandlerr ASAP would be good. The main problem right now is all those accounts we moved over to web portal that use this app have to remember to update settings in the app. Most have not done this, so they're looking up old content in Web Portal 😟

**Pull request merged**